packages/babel-plugin-i18n-calypso/src/index.js

```diff
@@ -0,0 +1,443 @@
+
+		if ( path.node.arguments.length > i && 'ObjectExpression' === path.node.arguments[i].type ) {
+			for ( const j in path.node.arguments[i].properties ) {
+				if ( 'ObjectProperty' === path.node.arguments[i].properties[j].type ) {
+					switch ( path.node.arguments[i].properties[j].key.name ) {
+						case 'context':
+							translation.msgctxt = path.node.arguments[i].properties[j].value.value;
+							break;
+						case 'comment':
+							translation.comments.extracted = path.node.arguments[i].properties[j].value.value;
+							break;
+					}
+				}
+			}
+		}
+
+		// Create context grouping for translation if not yet exists
+		const { msgctxt = '', msgid } = translation;
+		if ( ! strings[ filename ].hasOwnProperty( msgctxt ) ) {
+			strings[ filename ][ msgctxt ] = {};
+		}
+
+		strings[ filename ][ msgctxt ][ msgid ] = translation;
+	};
+
+	return {
+		visitor: {
+			CallExpression( path, state ) {
+				wordpress_i18n_CallExpression.apply( this, [ path, state ] );
+				i18n_calypso_CallExpression.apply( this, [ path, state ] );
+			},
+			Program: {
+				enter() {
+					strings[ this.file.opts.filename ] = {};
+				},
+				exit( path, state ) {
+					const { filename } = this.file.opts;
+					if ( isEmpty( strings[ filename ] ) ) {
+						delete strings[ filename ];
+						return;
+					}
+
+					// Sort translations by filename for deterministic output
+					const files = Object.keys( strings ).sort();
+
+					// Combine translations from each file grouped by context
+					const translations = reduce( files, ( memo, file ) => {
+						for ( const context in strings[ file ] ) {
+							// Within the same file, sort translations by line
+							const sortedTranslations = sortBy(
+								strings[ file ][ context ],
+								'comments.reference'
+							);
+
+							forEach( sortedTranslations, ( translation ) => {
+								const { msgctxt = '', msgid } = translation;
+								if ( ! memo.hasOwnProperty( msgctxt ) ) {
+									memo[ msgctxt ] = {};
+								}
+
+								// Merge references if translation already exists
+								if ( isSameTranslation( translation, memo[ msgctxt ][ msgid ] ) ) {
+									translation.comments.reference = uniq( [
+										memo[ msgctxt ][ msgid ].comments.reference,
+										translation.comments.reference,
+									].join( '\n' ).split( '\n' ) ).join( '\n' );
+								}
+
+								memo[ msgctxt ][ msgid ] = translation;
+							} );
+						}
+
+						return memo;
+					}, {} );
+
+					// Merge translations from individual files into headers
+					const data = merge( {}, baseData, { translations } );
+
+					// Ideally we could wait until Babel has finished parsing
+					// all files or at least asynchronously write, but the
+					// Babel loader doesn't expose these entry points and async
+					// write may hit file lock (need queue).
+					const compiled = po.compile( data );
+					writeFileSync( state.opts.output || DEFAULT_OUTPUT, compiled );
```

Comment:
Have you tried something like this instead?
```suggestion
					writeFileSync( state.opts.output || DEFAULT_OUTPUT, compiled, { flag: 'as' } );
```

I think `{ flag: 'a' }` would work too, or even going back to `writeFile` (async) by using the proper [mode flag](https://nodejs.org/api/fs.html#fs_file_system_flags).

Reply:
Thanks for that input. Might be an option but I disagree with writing the same file over and over again in the first place, so I chose another solution now.